hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/EndpointBase.java

```diff
@@ -63,6 +64,8 @@ public abstract class EndpointBase implements Auditor {
   private S3Auth s3Auth;
   @Context
   private ContainerRequestContext context;
+  @Inject
+  private OzoneConfiguration ozoneConfiguration;
```

Comment:
Instead of injecting the `ozoneConfiguration` here at the` EndpointBase` and redefining the default value for the _ozone.om.group.rights_, we can set the default value when we define the configuration at the _CDI_ producer either `Gateway.java` or `OzoneConfigurationHolder`.
https://github.com/apache/ozone/blob/a8808d1c3781627c40e0ed25d0bb4ec1e74e3de2/hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/Gateway.java#L70
or
https://github.com/apache/ozone/blob/a8808d1c3781627c40e0ed25d0bb4ec1e74e3de2/hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneConfigurationHolder.java#L41

Having the configuration default for group access set here, defines it for all injected `OzoneConfiguration` instances (endpoints, client producer - all s3 consumers).  Also, this is S3 specific, and does _not_ affect any other `RPCClient`.

Reply:
I would like it to be as close to request processing and to make the actual ACL setup as obvious as possible. This allow for more eyeballs looking at the code and its details and improves information density in the command processing code path vs. defaults being set transparently many layers down (as it was).
I agree that it is better to define this in the Configuration Holder as we never want the group policy to be set in any S3 context unless explicitly set. Updated the PR with the change.